Sources/Flows/OAuth2.swift

```diff
@@ -285,6 +285,12 @@ open class OAuth2: OAuth2Base {
 		if clientConfig.safariCancelWorkaround {
 			req.params["swa"] = "\(Date.timeIntervalSinceReferenceDate)" // Safari issue workaround
 		}
+		if clientConfig.useProofKeyForCodeExchange {
+			let codeVerifier = generateCodeVerifier()
+			req.params["code_challenge"] = generateCodeChallenge(verifier: codeVerifier)
+			req.params["code_challenge_method"] = "S256"
+			context.codeVerifier = codeVerifier
```

Comment:
I think it would be better to hide this generation in `context`. Have something like `context.generateCodeVerifier()` that internally generates the challenge and assigns internally to `context.codeVerifier`. Then read it out here and also have a read-only property `context.codeChallengeMethod` that returns "S256".

Reply:
Done